azurerm/internal/services/healthcare/parse/healthcare_service_test.go

```diff
@@ -0,0 +1,73 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestHealthcareServiceId(t *testing.T) {
+	testData := []struct {
+		Name     string
+		Input    string
+		Expected *HealthcareServiceId
+	}{
+		{
+			Name:     "Empty",
+			Input:    "",
+			Expected: nil,
+		},
+		{
+			Name:     "No Resource Groups Segment",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000",
+			Expected: nil,
+		},
+		{
+			Name:     "No Resource Groups Value",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/",
+			Expected: nil,
+		},
+		{
+			Name:     "Resource Group ID",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/foo/",
+			Expected: nil,
+		},
+		{
+			Name:     "Missing Service Value",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.HealthcareApis/services/",
+			Expected: nil,
+		},
+		{
+			Name:  "Healthcare Service ID",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.HealthcareApis/services/Service1",
+			Expected: &HealthcareServiceId{
+				Name:          "Service1",
+				ResourceGroup: "resGroup1",
+			},
+		},
+		{
+			Name:     "Wrong Casing",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.HealthcareApis/Services/",
```

Comment:
Should have a value on there also or will fail on missing value also, possibly hiding a condition?

Reply:
Indeed. Thank you for catching that!